astropy/coordinates/polarization.py

```diff
@@ -0,0 +1,252 @@
+from collections import namedtuple
+from contextlib import contextmanager
+from typing import Dict
+
+import numpy as np
+
+from astropy.utils import unbroadcast
+from astropy.utils.data_info import MixinInfo
+from astropy.utils.shapes import ShapedLikeNDArray
+
+__all__ = ["StokesCoord", "custom_stokes_symbol_mapping", "StokesSymbol"]
+
+
+StokesSymbol = namedtuple("StokesSymbol", ["symbol", "description"], defaults=[""])
+
+# This is table 29 in the FITS 4.0 paper
+FITS_STOKES_VALUE_SYMBOL_MAP = {
+    1: StokesSymbol("I", "Standard Stokes unpolarized"),
+    2: StokesSymbol("Q", "Standard Stokes linear"),
+    3: StokesSymbol("U", "Standard Stokes linear"),
+    4: StokesSymbol("V", "Standard Stokes circular"),
+    -1: StokesSymbol("RR", "Right-right circular: <RR*>"),
+    -2: StokesSymbol("LL", "Left-left circular: <LL*>"),
+    -3: StokesSymbol("RL", "Right-left cross-circular: Re(<RL*>))"),
+    -4: StokesSymbol("LR", "Left-right cross-circular: Re(<LR*>)=Im(<RL*>)"),
+    -5: StokesSymbol("XX", "X parallel linear: <XX*>"),
+    -6: StokesSymbol("YY", "Y parallel linear: <YY*>"),
+    -7: StokesSymbol("XY", "XY cross linear: Re(<XY*>)"),
+    -8: StokesSymbol("YX", "YX cross linear: Im(<XY*>)"),
+}
+
+STOKES_VALUE_SYMBOL_MAP = FITS_STOKES_VALUE_SYMBOL_MAP.copy()
+UNKNOWN_STOKES_VALUE = -99999
+
+
+@contextmanager
+def custom_stokes_symbol_mapping(
+    mapping: Dict[int, StokesSymbol], replace: bool = False
+):
+    """
+    Add a custom set of mappings from values to Stokes symbols.
+
+    Parameters
+    ----------
+    mappings
+        A list of dictionaries with custom mappings between values (integers)
+        and `.StokesSymbol` classes.
+    replace
+        Replace all mappings with this one.
+    """
+    global STOKES_VALUE_SYMBOL_MAP
+
+    original_mapping = STOKES_VALUE_SYMBOL_MAP.copy()
+    if not replace:
+        STOKES_VALUE_SYMBOL_MAP = {**original_mapping, **mapping}
```

Comment:
If it's an Enum, this can be combined as 

```python
STOKES_VALUE_SYMBOL_MAP = type("STOKES_VALUE_SYMBOL_MAP", (type(original_mapping), type(mapping)))
```